### PR TITLE
fix(youtube/settings): fix non functional back button in settings

### LINKED ISSUE #1511: bug: The back button at the top left corner for ReVanced, RYD and SponsorBlock settings is non-functional
### Type

Other

### Bug description

Clicking on the back button at the top left corner of the settings pages for ReVanced, Return Youtube Dislike and SponsorBlock does not do anything

### Steps to reproduce

1. Patch YouTube v17.49.37 with the latest release of ReVanced manager (v0.0.52) and all the patches
2. Open the patched app, navigate to either of the three settings pages and click on the back button

### Relevant log output

```
12-31 00:04:08.789  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: ViewPostIme pointer 0
12-31 00:04:08.864  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: ViewPostIme pointer 1
12-31 00:04:08.903  4547  4547 I DecorView: [INFO] isPopOver=false config=true
12-31 00:04:08.903  4547  4547 I DecorView: updateCaptionType: isFloating=false isApplication=true hasWindowDecorCaption=false this=DecorView@8841347[]
12-31 00:04:08.903  4547  4547 D DecorView: setCaptionType = 0, this = DecorView@8841347[]
12-31 00:04:08.903  4547  4547 I DecorView: getCurrentDensityDpi: from real metrics. densityDpi=420 msg=resources_loaded
12-31 00:04:08.903  4547  4547 I DecorView: setWindowBackground: isPopOver=false color=ff000000 d=android.graphics.drawable.ColorDrawable@68eb49d
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper: Couldn't set Toolbar title
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.TextView.setText(java.lang.CharSequence)' on a null object reference
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at app.revanced.integrations.settingsmenu.ReVancedSettingActivity.initializeSettings(ReVancedSettingActivity.java:77)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at com.google.android.libraries.social.licenses.LicenseActivity.onCreate(PG:1)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.Activity.performCreate(Activity.java:8578)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.Activity.performCreate(Activity.java:8557)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4147)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4312)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.os.Handler.dispatchMessage(Handler.java:106)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.os.Looper.loopOnce(Looper.java:226)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.os.Looper.loop(Looper.java:313)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at android.app.ActivityThread.main(ActivityThread.java:8741)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at java.lang.reflect.Method.invoke(Native Method)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
12-31 00:04:08.908  4547  4547 E revanced: ThemeHelper:         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
12-31 00:04:08.922  4547  4547 I MSHandlerLifeCycle: check: return. pkg=app.revanced.android.youtube parent=null callers=com.android.internal.policy.DecorView.setVisibility:4373 android.app.ActivityThread.handleResumeActivity:5463 android.app.servertransaction.ResumeActivityItem.execute:54 android.app.servertransaction.ActivityTransactionItem.execute:45 android.app.servertransaction.TransactionExecutor.executeLifecycleState:176
12-31 00:04:08.922  4547  4547 I MSHandlerLifeCycle: removeMultiSplitHandler: no exist. decor=DecorView@8841347[]
12-31 00:04:08.923  4547  4727 D NativeCustomFrequencyManager: [NativeCFMS] BpCustomFrequencyManager::BpCustomFrequencyManager()
12-31 00:04:08.926  4547  4547 D InsetsController: onStateChanged: InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 88 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(512, 0 - 568, 88), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 M 0,10.71963616907435 a 9.280363830925644,9.280363830925644 0 1,0 0,18.56072766185129 a 9.280363830925644,9.280363830925644 0 1,0 0,-18.56072766185129 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=84, center=Point(84, 84)}, RoundedCorner{position=TopRight, radius=84, center=Point(996, 84)}, RoundedCorner{position=BottomRight, radius=84, center=Point(996, 2316)}, RoundedCorner{position=BottomLeft, radius=84, center=Point(84, 2316)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(964, 0 - 1080, 88) rotation=0}, mSources= { InsetsSource: {mType=ITYPE_STATUS_BAR, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_NAVIGATION_BAR, mFrame=[0,2400][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_GESTURES, mFrame=[0,0][63,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_GESTURES, mFrame=[1017,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_MANDATORY_GESTURES, mFrame=[0,0][1080,120], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_MANDATORY_GESTURES, mFrame=[0,2348][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_DISPLAY_CUTOUT, mFrame=[0,0][-100000,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_DISPLAY_CUTOUT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_DISPLAY_CUTOUT, mFrame=[100000,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_DISPLAY_CUTOUT, mFrame=[0,100000][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_TAPPABLE_ELEMENT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_TAPPABLE_ELEMENT, mFrame=[0,0][0,0], mVisible=true, mInsetsRoundedCornerFrame=false} } host=app.revanced.android.youtube/com.google.android.libraries.social.licenses.LicenseActivity from=android.view.ViewRootImpl.setView:1700
12-31 00:04:08.927  4547  4547 I ViewRootImpl@73ded2e[LicenseActivity]: setView = com.android.internal.policy.DecorView@8841347 TM=true
12-31 00:04:08.927  4547  4547 I MSHandlerLifeCycle: removeMultiSplitHandler: no exist. decor=DecorView@8841347[LicenseActivity]
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: performTraversals params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} layoutInDisplayCutoutMode=shortEdges ty=BASE_APPLICATION wanim=0x1030309
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   fl=81810100
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   pfl=12020040
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   bhv=DEFAULT
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   fitSides= naviIconColor=0}
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: performTraversals mFirst=true windowShouldResize=true viewVisibilityChanged=false mForceNextWindowRelayout=false params={(0,0)(fillxfill) sim={adjust=resize forwardNavigation} layoutInDisplayCutoutMode=shortEdges ty=BASE_APPLICATION wanim=0x1030309
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   fl=81810100
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   pfl=12020040
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   bhv=DEFAULT
12-31 00:04:08.929  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]:   fitSides= naviIconColor=0}
12-31 00:04:08.934  4547  4547 D InsetsController: onStateChanged: InsetsState: {mDisplayFrame=Rect(0, 0 - 1080, 2400), mDisplayCutout=DisplayCutout{insets=Rect(0, 88 - 0, 0) waterfall=Insets{left=0, top=0, right=0, bottom=0} boundingRect={Bounds=[Rect(0, 0 - 0, 0), Rect(512, 0 - 568, 88), Rect(0, 0 - 0, 0), Rect(0, 0 - 0, 0)]} cutoutPathParserInfo={CutoutPathParserInfo{displayWidth=1080 displayHeight=2400 physicalDisplayWidth=1080 physicalDisplayHeight=2400 density={3.0} cutoutSpec={M 0,0 M 0,10.71963616907435 a 9.280363830925644,9.280363830925644 0 1,0 0,18.56072766185129 a 9.280363830925644,9.280363830925644 0 1,0 0,-18.56072766185129 Z @dp} rotation={0} scale={1.0} physicalPixelDisplaySizeRatio={1.0}}}}, mRoundedCorners=RoundedCorners{[RoundedCorner{position=TopLeft, radius=84, center=Point(84, 84)}, RoundedCorner{position=TopRight, radius=84, center=Point(996, 84)}, RoundedCorner{position=BottomRight, radius=84, center=Point(996, 2316)}, RoundedCorner{position=BottomLeft, radius=84, center=Point(84, 2316)}]}  mRoundedCornerFrame=Rect(0, 0 - 1080, 2400), mPrivacyIndicatorBounds=PrivacyIndicatorBounds {static bounds=Rect(964, 0 - 1080, 88) rotation=0}, mSources= { InsetsSource: {mType=ITYPE_STATUS_BAR, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_NAVIGATION_BAR, mFrame=[0,2400][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_GESTURES, mFrame=[0,0][63,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_GESTURES, mFrame=[1017,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_MANDATORY_GESTURES, mFrame=[0,0][1080,120], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_MANDATORY_GESTURES, mFrame=[0,2348][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_LEFT_DISPLAY_CUTOUT, mFrame=[0,0][-100000,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_DISPLAY_CUTOUT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_RIGHT_DISPLAY_CUTOUT, mFrame=[100000,0][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_DISPLAY_CUTOUT, mFrame=[0,100000][1080,2400], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_TOP_TAPPABLE_ELEMENT, mFrame=[0,0][1080,88], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_BOTTOM_TAPPABLE_ELEMENT, mFrame=[0,0][0,0], mVisible=true, mInsetsRoundedCornerFrame=false}, InsetsSource: {mType=ITYPE_IME, mFrame=[0,0][0,0], mVisibleFrame=[0,2400][1080,2400], mVisible=false, mInsetsRoundedCornerFrame=false} } host=app.revanced.android.youtube/com.google.android.libraries.social.licenses.LicenseActivity from=android.view.ViewRootImpl.relayoutWindow:9590
12-31 00:04:08.934  4547  4547 D BLASTBufferQueue: new BLASTBufferQueue, mNativeObject= 0xb400007ecc314d00 name= ViewRootImpl@73ded2e[LicenseActivity] sc.mNativeObject= 0xb400007e22b44dc0 caller= android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2789 android.view.ViewRootImpl.relayoutWindow:9635 android.view.ViewRootImpl.performTraversals:3759 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301 android.view.Choreographer$CallbackRecord.run:1309 android.view.Choreographer.doCallbacks:923 android.view.Choreographer.doFrame:852 android.view.Choreographer$FrameDisplayEventReceiver.run:1283
12-31 00:04:08.934  4547  4547 D BLASTBufferQueue: update, w= 1080 h= 2400 mNativeObject= 0xb400007ecc314d00 sc.mNativeObject= 0xb400007e22b44dc0 format= -1 caller= android.graphics.BLASTBufferQueue.<init>:76 android.view.ViewRootImpl.updateBlastSurfaceIfNeeded:2789 android.view.ViewRootImpl.relayoutWindow:9635 android.view.ViewRootImpl.performTraversals:3759 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665
12-31 00:04:08.935  4547  4547 I ViewRootImpl@73ded2e[LicenseActivity]: Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)0 dur=5 res=0x3 s={true 0xb400007e22ba0800} ch=true seqId=0
12-31 00:04:08.935  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: mThreadedRenderer.initialize() mSurface={isValid=true 0xb400007e22ba0800} hwInitialized=true
12-31 00:04:08.935  4547  4727 D OpenGLRenderer: eglCreateWindowSurface
12-31 00:04:08.941  4547  4547 D AbsListView:  in onLayout changed
12-31 00:04:08.941  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: reportNextDraw android.view.ViewRootImpl.performTraversals:4313 android.view.ViewRootImpl.doTraversal:2991 android.view.ViewRootImpl$TraversalRunnable.run:10665 android.view.Choreographer$CallbackRecord.run:1301 android.view.Choreographer$CallbackRecord.run:1309
12-31 00:04:08.941  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: Setup new sync id=0
12-31 00:04:08.941  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: Setting syncFrameCallback
12-31 00:04:08.941  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: registerCallbacksForSync syncBuffer=false
12-31 00:04:08.942  4547  4795 D ViewRootImpl@73ded2e[LicenseActivity]: Received frameDrawingCallback syncResult=0 frameNum=1.
12-31 00:04:08.942  4547  4795 D ViewRootImpl@73ded2e[LicenseActivity]: Setting up sync and frameCommitCallback
12-31 00:04:08.943  4547  4727 D BLASTBufferQueue: [ViewRootImpl@73ded2e[LicenseActivity]#7](f:0,a:0) onFrameAvailable the first frame is available
12-31 00:04:08.944  4547  4727 D ViewRootImpl@73ded2e[LicenseActivity]: Received frameCommittedCallback lastAttemptedDrawFrameNum=1 didProduceBuffer=true
12-31 00:04:08.944  4547  4727 D OpenGLRenderer: CFMS:: SetUp Pid : 4547    Tid : 4727
12-31 00:04:08.944  4547  4727 W Parcel  : Expecting binder but got null!
12-31 00:04:08.944  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: onSyncComplete
12-31 00:04:08.945  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: setupSync seqId=0 mSyncId=0 fn=1 caller=android.view.ViewRootImpl$$ExternalSyntheticLambda11.accept:6 android.window.SurfaceSyncer.lambda$setupSync$1$android-window-SurfaceSyncer:128 android.window.SurfaceSyncer$$ExternalSyntheticLambda1.accept:8 android.window.SurfaceSyncer$SyncSet.checkIfSyncIsComplete:382 android.window.SurfaceSyncer$SyncSet.markSyncReady:359 android.window.SurfaceSyncer.markSyncReady:151 android.view.ViewRootImpl.performTraversals:4373
12-31 00:04:08.945  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: reportDrawFinished seqId=0 mSyncId=-1 fn=1 mSurfaceChangedTransaction=0xb400007ec1623000
12-31 00:04:08.945  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: MSG_WINDOW_FOCUS_CHANGED 0 0
12-31 00:04:08.964  4547  4547 I ViewRootImpl@73ded2e[LicenseActivity]: MSG_WINDOW_FOCUS_CHANGED 1 0
12-31 00:04:08.964  4547  4547 D ViewRootImpl@73ded2e[LicenseActivity]: mThreadedRenderer.initializeIfNeeded()#2 mSurface={isValid=true 0xb400007e22ba0800}
12-31 00:04:08.964  4547  4547 D InputMethodManager: startInputInner - Id : 0
12-31 00:04:08.964  4547  4547 I InputMethodManager: startInputInner - mService.startInputOrWindowGainedFocus
12-31 00:04:09.349  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: handleAppVisibility mAppVisible = true visible = false
12-31 00:04:09.355  4547  4727 D OpenGLRenderer: setSurface called with nullptr
12-31 00:04:09.355  4547  4727 D OpenGLRenderer: setSurface() destroyed EGLSurface
12-31 00:04:09.355  4547  4727 D OpenGLRenderer: destroyEglSurface
12-31 00:04:09.357  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]: performTraversals mFirst=false windowShouldResize=false viewVisibilityChanged=true mForceNextWindowRelayout=false params=null
12-31 00:04:09.360  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)8 dur=2 res=0x2 s={false 0x0} ch=true seqId=0
12-31 00:04:09.361  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: stopped(true) old = false
12-31 00:04:09.361  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]: WindowStopped on app.revanced.android.youtube/com.google.android.apps.youtube.app.settings.SettingsActivity set to true
12-31 00:04:09.367  4547  4547 I MSHandlerLifeCycle: removeMultiSplitHandler: no exist. decor=DecorView@62f332[SettingsActivity]
12-31 00:04:09.371  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]: performTraversals mFirst=false windowShouldResize=false viewVisibilityChanged=false mForceNextWindowRelayout=false params={(0,0)(fillxfill) sim={adjust=pan} ty=BASE_APPLICATION wanim=0x1030309 sysuil=true
12-31 00:04:09.371  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]:   fl=81810100
12-31 00:04:09.371  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]:   pfl=12020040
12-31 00:04:09.371  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]:   bhv=DEFAULT
12-31 00:04:09.371  4547  4547 D ViewRootImpl@c402b48[SettingsActivity]:   fitSides= naviIconColor=0}
12-31 00:04:09.374  4547  4547 I ViewRootImpl@c402b48[SettingsActivity]: Relayout returned: old=(0,0,1080,2400) new=(0,0,1080,2400) req=(1080,2400)8 dur=3 res=0x2 s={false 0x0} ch=false seqId=0
12-31 00:04:11.450  4547  4547 I ViewRootImpl@73ded2e[LicenseActivity]: ViewPostIme pointer 0
12-31 00:04:11.518  4547  4547 I ViewRootImpl@73ded2e[LicenseActivity]: ViewPostIme pointer 1
```

### Screenshots or videos

_No response_

### Solution

_No response_

### Additional context

The log output that I have included is of me clicking on the ReVanced option in the settings page (at 00:04:08.789) and then clicking on the back button (at 00:04:11.450).

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

### PR DESCRIPTION
**Pull request by LisoUseInAIKyrios**
Fixed the back button not working.

[integration changes](https://github.com/revanced/revanced-integrations/pull/404)

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/sponsorblock/resource/patch/SponsorBlockResourcePatch.kt [hunk 1]
```diff
@@ -27,12 +27,12 @@ class SponsorBlockResourcePatch : ResourcePatch {
...
         SettingsPatch.addPreference(
             Preference(
-                StringResource("sb_settings", "SponsorBlock"),
+                StringResource("revanced_sponsorblock_settings_title", "SponsorBlock"),
                 StringResource("revanced_sponsorblock_settings_summary", "SponsorBlock related settings"),
                 Preference.Intent(
                     youtubePackage,
                     "sponsorblock_settings",
-                    "com.google.android.libraries.social.licenses.LicenseActivity"
+                    "app.revanced.integrations.settingsmenu.ReVancedSettingActivity"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/sponsorblock/resource/patch/SponsorBlockResourcePatch.kt:35]: A constant may be provided for this string.

**Review by oSumAtrIX — COMMENTED** [review]
Has this been confirmed to work on non-root as well as root?

**Comment by inotia00** [comment]
If I remember correctly, for some reason we couldn't declare a new activity in the AndroidManifest in the root environment

For the same reason, TeamVanced also uses LicenseActivity in root builds

**Comment by 0xrxL** [comment]
> In that case it should use the existing strategy of modifying a built in activity. I've reverted to that for this PR.

Then, in the current state, this fix it's functional for both root/non-root?

**Comment by oSumAtrIX** [comment]
@LisoUseInAIKyrios Have you tried if this works on non root before reverting the code?

**Comment by oSumAtrIX** [comment]
You can root an AVD by using this

**Comment by oSumAtrIX** [comment]
> PlayStore is not available in the emulator

You need an image with GMS. You can select one in AVD Manager.

> It appears a root install on the emulator may not be possible.

I have previously successfully updated YouTube on an emulator and mounted a patched app with root.

**Comment by oSumAtrIX** [comment]
@LisoUseInAIKyrios Yeah, it may not be as easy to use the emulator, though it is preferable not to use your daily driver for development and instead rely on an emulator to avoid problems.

**Comment by johnconner122** [comment]
Try `WSA (Windows Subsystem for Android)`, it's performance is pretty good.

**Comment by inotia00** [comment]
yeah.. that's what i was referring to

In the root environment, Vanced and ReVanced are installed with `mount -o bind`

but when we install with this method, stock YouTube's signature and manifest are used

**Comment by 0xrxL** [comment]
> yeah.. that's what i was referring to
> 
> In the root environment, Vanced and ReVanced are installed with `mount -o bind`
> 
> but when we install with this method, stock YouTube's signature and manifest are used

Link the solution from your repo at this point.

**Comment by inotia00** [comment]
> Link the solution from your repo at this point.

we don't have to do that
current PR works great enough to be merged immediately

**Pull request merged**